dallinger/command_line.py

```diff
@@ -318,6 +319,20 @@ def deploy_sandbox_shared_setup(verbose=True, app=None, web_procs=1, exp_config=
     (id, tmp) = setup_experiment(debug=False, verbose=verbose, app=app,
                                  exp_config=exp_config)
 
+    # Register the experiment on OSF
+    # Load configuration.
+    config = get_config()
+    if not config.ready:
+        config.load_config()
+
+    try:
+        config.get('osf_access_token')
+        log("Preregistering experiment on OSF...")
+        registration.register(id, snapshot=None)
+
+    except Exception:
+        pass
```

Comment:
Thanks. Here, there *is* a particular issue that we're anticipating. The OSF integration is optional, and so if the user does not specify keys we want to handle that situation gracefully and proceed with launching the experiment without the OSF registration.

I've changed this to `except KeyError`, which is the exception raised by the configuration object when a config parameter cannot be found.